hail/src/main/scala/is/hail/expr/ir/SpecializedArrayBuilders.scala

```diff
@@ -9,8 +9,9 @@ import is.hail.utils.BoxedArrayBuilder
 
 import scala.reflect.ClassTag
 
-class StagedArrayBuilder(val elt: SingleCodeType, val eltRequired: Boolean, mb: EmitMethodBuilder[_], len: Code[Int]) {
+class StagedArrayBuilder(cb: EmitCodeBuilder, val elt: SingleCodeType, val eltRequired: Boolean, len: Int) {
```

Comment:
I don't think it's the right move for this class to keep hold of a code-builder. It should just need a class builder to define the layout and methods of an array. Really, it should just need a module builder.
Seems the normal plan is to thread one through method calls.